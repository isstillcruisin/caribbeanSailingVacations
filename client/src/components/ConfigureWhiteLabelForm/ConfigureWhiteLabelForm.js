import React, {Component} from 'react'
import { Button, Form, Col, Card } from 'react-bootstrap'
import ImageUploader from '../ImageUploader'


class ConfigureWhiteLabelForm extends Component {
  render() {
    return <div style={{color: 'black', padding: '10px'}}>
      <Card>
        <Card.Header>Contact Information</Card.Header>
        <Card.Body>
          <Form>
            <Form.Row>
              <Form.Group as={Col} controlId="formGridTitle">
                <Form.Label>White Label Title</Form.Label>
                <Form.Control 
                  placeholder="White Label Title" 
                  name='title'
                  onChange={this.props.handleInputChange}
                  value={this.props.whiteLabel.title}
                />
              </Form.Group>
            </Form.Row>
            <Form.Row>
              <Form.Group as={Col} controlId="formGridTitle">
                <Form.Label>'About Us' Text <i>(This is the text that will show on the ABOUT section of an E-Brochure under this white label.)</i>
                </Form.Label>
                <Form.Control 
                  placeholder="'About Us' Text" 
                  name='aboutText'
                  onChange={this.props.handleInputChange}
                  value={this.props.whiteLabel.aboutText}
                  as="textarea" rows="3"
                />
              </Form.Group>
            </Form.Row>
            <Form.Row>
              <Form.Group as={Col} controlId="formGridCompanyName">
                <Form.Label>Company Name</Form.Label>
                <Form.Control 
                  placeholder="Company Name" 
                  name='companyName'
                  onChange={this.props.handleInputChange}
                  value={this.props.whiteLabel.companyName}
                />
              </Form.Group>
            </Form.Row>
            <Form.Group>
              <Form.Label>Street address</Form.Label>
              <Form.Control 
                placeholder="Business street address"
                name='streetAddress' 
                onChange={this.props.handleInputChange}
                value={this.props.whiteLabel.streetAddress} 
              />
            </Form.Group>
            <Form.Row>
              <Form.Group as={Col} controlId="formGridCity">
                <Form.Label>City</Form.Label>
                <Form.Control 
                  placeholder="City" 
                  name='city'
                  onChange={this.props.handleInputChange}
                  value={this.props.whiteLabel.city}
                />
              </Form.Group>

              <Form.Group as={Col} controlId="formGridState">
                <Form.Label>State</Form.Label>
                <Form.Control
                  placeholder="State" 
                  name='state'
                  onChange={this.props.handleInputChange}
                  value={this.props.whiteLabel.state}
                />
              </Form.Group>

              <Form.Group as={Col} controlId="formGridZip">
                <Form.Label>Zip Code</Form.Label>
                <Form.Control
                  placeholder="Zip Code" 
                  name='zipCode'
                  onChange={this.props.handleInputChange}
                  value={this.props.whiteLabel.zipCode}
                />
              </Form.Group>
            </Form.Row>
            <Form.Row>
              <Form.Group as={Col} controlId="formGridCountry">
                <Form.Label>Country</Form.Label>
                <Form.Control 
                  placeholder="Country" 
                  name='country'
                  onChange={this.props.handleInputChange}
                  value={this.props.whiteLabel.country}
                />
              </Form.Group>
            </Form.Row>
            <Form.Row>
              <Form.Group>
                <ImageUploader
                  imgs={this.props.whiteLabel.logoUrl ? [this.props.whiteLabel.logoUrl] : []}
                  showDropZone={!this.props.whiteLabel.logoUrl}
                  onChange={this.props.handleInputChange}
                  setUrls={this.handleSetUrls}
                  placeholder='Drop a logo image here, or click to select a file to upload. The image will be scaled to fit in a rectangle of 200px wide, 100px tall.'
                  multiple={false}
                  width={200}
                  height={100}
                />
              </Form.Group>
            </Form.Row>
            <Form.Row>
              <Form.Group as={Col} controlId="formGridSubmitAddress">
                <Button
                  onClick={this.props.handleSaveWhiteLabel}
                  type="submit"
                  className="btn btn-lg"
                  disabled={this.props.saved}
                > 
                  {this.props.saved ? 'Saved' : 'Save'}
                </Button>
              </Form.Group>
            </Form.Row>
          </Form>
        </Card.Body>
      </Card>
    </div>
  }

  handleSetUrls = urlArray => {
    this.props.handleInputChange({ target: { name: 'logoUrl', value: urlArray[0] }})
  };
}

export default ConfigureWhiteLabelForm
