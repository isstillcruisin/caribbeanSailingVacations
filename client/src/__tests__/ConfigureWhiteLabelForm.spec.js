import React from "react";
import { create } from "react-test-renderer";
import ConfigureWhiteLabelForm from '../components/ConfigureWhiteLabelForm';
import { MemoryRouter } from 'react-router';

describe("ConfigureWhiteLabelForm component", () => {
  test("Matches the snapshot", () => {
    const form = create(
      <MemoryRouter>
        <ConfigureWhiteLabelForm 
          whiteLabel={
            {
              streetAddress: '1 Test Lane',
              name: 'test',
              ebrochures: [
                {
                  name: 'test brochure',
                  _id: '1',
                }
              ],
            }
          }
        />
      </MemoryRouter>
    );
    expect(form.toJSON()).toMatchSnapshot();
  });
});