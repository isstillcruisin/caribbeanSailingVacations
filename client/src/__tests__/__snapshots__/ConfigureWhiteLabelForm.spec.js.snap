// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ConfigureWhiteLabelForm component Matches the snapshot 1`] = `
<div
  style={
    Object {
      "color": "black",
    }
  }
>
  <form
    className=""
  >
    <div
      className="add-white-label-form form-group"
    >
      <label
        className="form-label"
      >
        White-Label Name
      </label>
      <input
        className="form-control form-control"
        disabled={true}
        id="white-label-name"
        name="whiteLabelName"
        required={true}
        type="text"
        value="test"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
      >
        Phone number
      </label>
      <input
        className="form-control"
        name="phoneNumber"
        placeholder="Phone number"
      />
    </div>
    <div
      className="form-group"
    >
      <label
        className="form-label"
      >
        Street address
      </label>
      <input
        className="form-control"
        name="streetAddress"
        placeholder="Business street address"
        value="1 Test Lane"
      />
    </div>
    <div
      className="form-row"
    >
      <div
        className="form-group col"
      >
        <label
          className="form-label"
          htmlFor="formGridCity"
        >
          City
        </label>
        <input
          className="form-control"
          id="formGridCity"
          name="city"
          placeholder="City"
        />
      </div>
      <div
        className="form-group col"
      >
        <label
          className="form-label"
          htmlFor="formGridState"
        >
          State
        </label>
        <input
          className="form-control"
          id="formGridState"
          name="state"
          placeholder="State"
        />
      </div>
      <div
        className="form-group col"
      >
        <label
          className="form-label"
          htmlFor="formGridZip"
        >
          Zip Code
        </label>
        <input
          className="form-control"
          id="formGridZip"
          name="zipCode"
          placeholder="Zip Code"
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <label
          className="form-label"
        >
          Country
        </label>
        <input
          className="form-control"
          name="country"
          placeholder="Country"
        />
      </div>
    </div>
    <div
      className="row"
    >
      <div
        className="col"
      >
        <button
          className="btn btn-lg btn btn-primary"
          disabled={false}
          type="submit"
        >
          Save Address And Phone Number
        </button>
      </div>
    </div>
  </form>
  <table
    className="table table-striped table-bordered table-hover"
  >
    <thead>
      <tr>
        <th>
          Name
        </th>
        <th>
          View
        </th>
        <th>
          Edit
        </th>
      </tr>
    </thead>
    <tbody>
      <tr>
        <td>
          test brochure
        </td>
        <td>
          <a
            className="e-brochure-link btn btn-primary"
            href="/e-brochure/1"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            View E-Brochure
          </a>
        </td>
        <td>
          <a
            className="edit-e-brochure-link btn btn-primary"
            href="/e-brochure/1/edit"
            onClick={[Function]}
            onKeyDown={[Function]}
          >
            Edit E-Brochure
          </a>
        </td>
      </tr>
    </tbody>
  </table>
  <a
    className="add-e-brochure-link btn btn-primary"
    href="/white-label/test/new-e-brochure"
    onClick={[Function]}
    onKeyDown={[Function]}
  >
    Add New E-Brochure
  </a>
</div>
`;
